pkg/service/utgen/injector_test.go

```diff
@@ -0,0 +1,358 @@
+function calculate() {
+    return 42;
+}`,
+			newImports: []string{
+				"import { sqrt } from './math';",
+			},
+			expected: `import { sqrt } from './math';
+/* This is a comment */
+function calculate() {
+    return 42;
+}`,
+			expectedDiff: 1,
+		},
+		{
+			name: "Handling empty lines and comments",
+			content: `import { isEven } from './math';
+// A comment explaining the code
+const { sum } = require('./utils');
+
+function calculate() {
+    return sum(2, 3);
+}`,
+			newImports: []string{
+				"import { divide } from './math';",
+				"const { multiply } = require('./utils');",
+			},
+			expected: `import { isEven } from './math';
+// A comment explaining the code
+const { sum } = require('./utils');
+import { divide } from './math';
+const { multiply } = require('./utils');
+
+function calculate() {
+    return sum(2, 3);
+}`,
+			expectedDiff: 2,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			updatedContent, diff, err := injector.updateJavaScriptImports(tt.content, tt.newImports)
+			if err != nil {
+				t.Fatalf("Unexpected error: %v", err)
+			}
+			if strings.TrimSpace(updatedContent) != strings.TrimSpace(tt.expected) {
+				t.Errorf("Expected:\n%s\nGot:\n%s", tt.expected, updatedContent)
+			}
+			if diff != tt.expectedDiff {
+				t.Errorf("Expected diff: %d, Got: %d", tt.expectedDiff, diff)
+			}
+		})
+	}
+}
+
+func TestUpdatePythonImports(t *testing.T) {
+	injector := &Injector{}
+
+	tests := []struct {
+		name         string
+		content      string
+		newImports   []string
+		expected     string
+		expectedDiff int
+	}{
+		{
+			name: "No new imports",
+			content: `from math import sqrt
+
+def func():
+    return sqrt(4)
+`,
+			newImports: []string{},
+			expected: `from math import sqrt
+
+def func():
+    return sqrt(4)
+`,
+			expectedDiff: 0,
+		},
+		{
+			name: "New imports with duplicates",
+			content: `from math import sqrt
+
+def func():
+    return sqrt(4)
+`,
+			newImports: []string{
+				"from math import sqrt",   // Duplicate
+				"from math import divide", // New import
+			},
+			expected: `from math import sqrt
+from math import divide
+
+def func():
+    return sqrt(4)
+`,
+			expectedDiff: 1,
+		},
+		{
+			name: "No existing imports, adding new imports",
+			content: `def func():
+    return sum(2, 3)
+`,
+			newImports: []string{
+				"import os",
+			},
+			expected: `import os
+def func():
+    return sum(2, 3)
+`,
+			expectedDiff: 1,
+		},
+		{
+			name: "Adding imports to content with no imports",
+			content: `# This is a comment
+def func():
+    return sum(2, 3)
+`,
+			newImports: []string{
+				"import os",
+			},
+			expected: `import os
+# This is a comment
+def func():
+    return sum(2, 3)
+`,
+			expectedDiff: 1,
+		},
+		{
+			name: "Handling empty lines and comments",
+			content: `from math import sqrt
+# A comment explaining the code
+from random import randint # checking coverage for file - do not remove
+
+def func():
```

Comment:
```
# checking coverage for file - do not remove
```
the code itself is removed ...this was used to handle file imports

Reply:
In the original logic, there was a focus on merging imports that had the same module name. For example, if there were two lines like from math import sqrt and from math import pow, the original code would merge them into a single line: from math import sqrt, pow. This merging was done while ensuring that any special comments (like # checking coverage for file - do not remove) were preserved. The logic was also built to handle cases where imports were annotated with specific comments and would make sure that these annotations remained intact.

The new commonImport approach, while still compatible with this functionality, simplifies the process. Instead of merging import statements, the newImport entries are appended to the existing ones, keeping them distinct. This means if you have both from math import sqrt and from math import pow in the new logic, they would remain as separate lines rather than being merged. However, the core functionality of preserving formatting and special comments is still maintained, and imports with annotations are left unchanged.

Reply:
the issue in this would be when user runs this they would see a lot of unneccesarry import lines that either they will manually merge. Thus I believe the earlier approach was better

Reply:
Thank you for your patience and detailed feedback! Based on you suggestions, I need to merge the imports differently depending on the language. For example, in Python, I should handle imports like from math import sqrt and from math import other by combing them into a single line like 'from math import sqrt, other'. This would reduce redundancy and avoid unnecessary import lines, ensuring the code remains clean and efficient.

Also, at the moment, I’ve only adjusted the logic to handle merging imports—are there any other areas you think need changes or further improvements?